python/cudf_polars/cudf_polars/dsl/utils/aggregations.py

```diff
@@ -0,0 +1,260 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utilities for rewriting aggregations."""
+
+from __future__ import annotations
+
+import itertools
+from functools import partial
+from typing import TYPE_CHECKING
+
+import pyarrow as pa
+
+import pylibcudf as plc
+
+from cudf_polars.dsl import expr, ir
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Generator, Iterable, Sequence
+
+    from cudf_polars.typing import Schema
+
+__all__ = [
+    "apply_pre_evaluation",
+    "decompose_aggs",
+    "decompose_single_agg",
+]
+
+
+def decompose_single_agg(
+    named_expr: expr.NamedExpr,
+    name_generator: Generator[str, None, None],
+    *,
+    is_top: bool,
+) -> tuple[list[expr.NamedExpr], expr.NamedExpr, bool]:
+    """
+    Decompose a single named aggregation.
+
+    Parameters
+    ----------
+    named_expr
+        The named aggregation to decompose
+    name_generator
+        Generator of unique names for temporaries introduced during decomposition.
+    is_top
+        Is this the top of an aggregation expression?
+
+    Returns
+    -------
+    tuple
```

Comment:
Super-Nit: I don't know if it's "correct" convention, but I find it easier to read a "Returns" block when the elements of the tuple are split into separate descriptions.

Reply:
For numpydoc, it's one type / description per element of the tuple when returning multiple values: https://numpydoc.readthedocs.io/en/latest/format.html#returns